tools/explorer/lib/ontology_wrapper.py

```diff
@@ -0,0 +1,137 @@
+"""
+Ontology wrapper class for ontology explorer exposing certain functionality
+of the Digital Buildings Ontology.
+"""
+from typing import List
+
+from yamlformat.validator.entity_type_lib import EntityType
+from yamlformat.validator.entity_type_manager import EntityTypeManager
+from yamlformat.validator.presubmit_validate_types_lib import ConfigUniverse
+
+from lib.model import StandardField
+from lib.model import EntityTypeField
+from lib.model import Match
+
+class OntologyWrapper(object):
+  """Class providing an interface to do lookups on a DigitalBuildings ontology.
+
+     Attributes:
+          universe: A ConfigUniverse object detailing the various universes in
+            the ontology
+          manager: An EntityTypeManager object to find greatest common subsets
+            of fields between entity types and complete lists of inheritied
+            fields for an entity type. This is primarily used for
+            _CreateMatch().
+
+     Returns:
+          An instance of Ontology class.
+  """
+
+  # TODO(travis):Validate that universe has types expanded and fast-fail if not
+  def __init__(self, universe: ConfigUniverse):
+    """Init.
+
+      Args:
+          universe: an instantiated ConfigUniverse Object with inherited fields
+          expanded.
+    """
+    super().__init__()
+    self.universe = universe
+    self.manager = EntityTypeManager(self.universe)
+
+  def GetFieldsForTypeName(
+      self,
+      namespace: str,
+      entity_type_name: str,
+      required_only: bool = False
+  ) -> List[EntityTypeField]:
+    """Gets a list of fields for a given typename within a namespace.
+
+    Args:
+       namespace: the name of the namespace as a string.
+       entity_type_name: the name of the entity type as a string.
+       required_only: when true will return only required fields for a given
+         type.
+
+    Returns:
+            result_fields: a list of StandardField tuples
+    """
+    entity_type = self.universe.entity_type_universe.GetEntityType(
+        namespace,
+        entity_type_name
+    )
+    #entity_type_lib.FieldParts NamedTuple to EntityTypeField object.
+    entity_type_fields = [
+        EntityTypeField(
+            namespace_name=field.field.namespace,
+            standard_field_name=field.field.field,
+            is_optional=field.optional,
+            increment=field.field.increment
+        )
+        for field in entity_type.GetAllFields().values()
+    ]
```

Comment:
this will not pass the internal linter please re-write

Reply:
Is it the constructor or the list comprehension that has linting failures?  The constructor could span fewer lines if you remove the explicit argument names, but if it's the list comprehension, it might be better to change it to a for block that appends to the list.

Reply:
The linter is failing on the naming of field.field.field. This has been changed to qualified_field.field.field which should pass the linter.